### ISSUE TITLE
Performance improvements for reindex operation.

### ISSUE THREAD

**Issue by davidkellerman** [nxus/searcher#17]
Reindex (after deleting ElasticSearch index) for OSHA inspections takes 4-6 hours for 140,000 records. Some of this might not be ElasticSearch – there's a preprocessor method that does a WL find() to essentially do a "join" with a related record. However, it looks like most of it is ES.

Question is whether there are specialized ES functions that would allow a more efficient bulk rebuild of the index.

**Comment by loppear** [comment]
Yeah, our single-document-at-a-time reindex isn't going to cut it long term.

1. We can use the [Bulk API](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-bulk.html), using the `index` action (as this will update for ids that already exist (though we may decide that wiping the index first is appropriate anyways, see next)
2. Need to think about whether we want to support mapping definition change reindexing in addition to data change reindexing. This would avoid resending all the records when the reason to reindex is just to fix the mapping schema. Downside is that this is awkward to integrate with the waterline connection approach, though doing this directly with the HTTP API behind the scenes may just work, since the steps I've used (with raw http scripts) are:
   1. Create a new temporary index with the new mapping definition
   1. Reindex the current index to the new one
   1. Drop the current index
   1. Reindex the temporary index to the current one.
